### ISSUE TITLE
some simultaneous OSC messages don't get through

### ISSUE THREAD

**Issue by daveriedstra** [padenot/monome-rs#3]
First, thanks for the useful crate, I'm quite enjoying it.

When sending a larger number of simultaneous messages to an arc, sometimes they don't all get there. For instance, when running the following, encoders three and four don't always light up as expected. Uncommenting the `sleep()` lines works around the issue. The same behaviour is not present in the equivalent puredata code.

```
extern crate monome;
use monome::Monome;

fn main() {
    let mut arc = Monome::new("/test".to_string()).unwrap();
    for i in 0..4 {
        arc.ring_all(i, 0);
    }

    make_dial(&mut arc, 0);
    // std::thread::sleep(std::time::Duration::from_millis(10));
    arc.ring_all(1, 6);
    // std::thread::sleep(std::time::Duration::from_millis(10));
    make_dial(&mut arc, 2);
    // std::thread::sleep(std::time::Duration::from_millis(10));
    make_dial(&mut arc, 3);
    std::thread::sleep(std::time::Duration::from_millis(10));
}

fn make_dial(arc: &mut Monome, enc: usize) {
    for i in 0..8 {
        arc.ring_set(enc, i * 8, 7);
    }
    for i in 0..6 {
        arc.ring_set(enc, i + 2, 15);
    }
}
```

![image](https://user-images.githubusercontent.com/3504098/103371254-fe7eab80-4a9c-11eb-9843-b462e22a0df0.png)

**Comment by padenot** [comment]
Thanks for the report, fixed in ba0bc0a, and subsequently released in version [`1.1.1`](https://crates.io/crates/monome-rs/1.1.1) with this fix.

The bug was that this was using a channel with a rather small limit of 16 elements to communicate between the application code and the device communication code. Your code was sending too many messages for this 16-element limit.

This limit had no reason to exist, clearly your code snippet is valid, so I switched to an unbounded buffer instead of just bumping the buffer size.

Happy to hear you enjoy this crate, and let me know if you find other bugs!

**Comment by daveriedstra** [comment]
I was wondering if it was something like that, I ran into something similar myself. Thanks for the quick fix!
